### DISCUSSION TITLE
ICN2037BP and FM6124 - Are they the same?

### DISCUSSION THREAD

**Discussion by Speedy-VI** [mrcodetastic/ESP32-HUB75-MatrixPanel-DMA#649] [Q&A]
Does anyone know if the ICN2037BP and the FM6124 LED driver chips are functionally the same? I have a panel that has the ICN2037BP chips that works with my Adafruit Matrix Portal S3. Adafruit sells a P2 64x64 panel that the product photo of the back shows ICN2037BP chips. I found these same Corman panels on Ali Express for $12.50 each (vs $50 each from Adafruit).

Coreman told me they used to sell Adafruit P2 64x64 panels with ICN2037BP chips but the last few batches have FM6124 chips that they say are identical. They also said they have P2 128x64 panels with 5 address lines and FM6124 chips. I have been searching for P2 128x64 panels that will work with the Matrix Portal S3 and think the Coreman panels may work but I need confirmation that FM6124 and ICN2037BP are interchangeable. Thanks!

**Comment by hijoh** [comment]
I tried ICN2037 is OK
But FM6127 has some problems.

**Reply by Speedy-VI** [reply]
I am by no means an expert but based on scrolling through hzeller's rpi-rgb-led-matrix GitHub discussions, the FM6127 is significantly different than the FM6124. His library supports FM6127 but it's for Raspberry Pi only. I don't think a Matrix Portal can control a panel with FM6127 driver chips.

**Comment by softhack007** [comment]
I have a 64x64 with ICN2037BP,  and another 128x64 with FM6124. They both work for me with the standard SHIFTREG driver, just needed to set "mxconfig.clkphase = false" to get rid of pixel flickering.

Not sure about "interchangeable" - you  probably can attach either one of them to your esp32 without recompiling the firmware. But exchanging one panel in a chain may work or not work.

**Reply by Speedy-VI** [reply]
Thanks for your input. I ended up buying 2 x P2 64x64 panels from Adafruit with  ICN2037BP driver chips and 4 x P2 64x64 panels from Coreman with FM6124 driver chips. The Coreman panels work fine with the Matrix Portal S3. Chaining the 2 Adafruit panels works, and chaining 2 Coreman panels works. Next, I will try chaining an Adafruit panel and a Coreman panel. Fingers crossed!

**Reply by pelayogarciatamargo-ai** [reply]
Hi! I need some help. I have an 128x64 with FM6124 and Matrix Portal S3. Can i ask you some questions? I'm desperate :,_____________)
